draft-bbs-signatures.md

```diff
@@ -473,52 +495,80 @@ Procedure:
 
 5. if W is INVALID, return INVALID
 
-6. generators =  (H_s || H_d || H_1 || ... || H_L)
+6. gen_octets =  (H_s || H_d || H_1 || ... || H_L)
+
+7. domain_prime = (PK || L || gen_octets || Ciphersuite_ID || header)
 
-7. domain = hash_to_scalar((PK || L || generators || Ciphersuite_ID || header), 1)
+8. domain = hash_to_scalar(domain_prime, 1)
 
-8. B = P1 + H_s * s + H_d * domain + H_1 * msg_1 + ... + H_L * msg_L
+9. B = P1 + H_s * s + H_d * domain + H_1 * msg_1 + ... + H_L * msg_L
 
-9. if e(A, W + P2 * e) * e(B, -P2) != Identity_GT, return INVALID
+10. if e(A, W + P2 * e) * e(B, -P2) != Identity_GT, return INVALID
 
-10. return VALID
+11. return VALID
 ```
 
 ### ProofGen
 
 This operation computes a zero-knowledge proof-of-knowledge of a signature, while optionally selectively disclosing from the original set of signed messages. The "prover" may also supply a presentation header, see [presentation header selection](#presentation-header-selection) for more details.
 
-If an application chooses to pass the indexes of the generators instead, then it will also need to pass the indexes of the generators corresponding to the revealed messages.
+The messages supplied in this operation MUST be in the same order as when supplied to [Sign](#sign). To specify which of those messages will be disclosed, the prover can supply the list of indexes (`disclosedIndexes`) that the disclosed messages have in the array of signed messages. Each element in `disclosedIndexes` MUST be a non-negative integer, in the range from 1 to `length(messages)`.
 
 ```
-proof = ProofGen(PK, signature, header, ph, (msg_1,..., msg_L), (H_1,..., H_L), RevealedIndexes)
+proof = ProofGen(PK, signature, header, ph, disclosedIndexes, messages)
 
 Inputs:
 
-- PK (REQUIRED), an octet string of the form outputted by the SkToPk operation.
-- signature (REQUIRED), an octet string of the form outputted by the Sign operation.
-- header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
+- PK (REQUIRED), an octet string of the form outputted by the SkToPk
+                 operation.
+- signature (REQUIRED), an octet string of the form outputted by the
+                        Sign operation.
+- header (OPTIONAL), an octet string containing context and application
+                     specific information. If not supplied, it defaults
+                     to an empty string.
 - ph (OPTIONAL), octet string containing the presentation header.
-- msg_1,..., msg_L (OPTIONAL), octet strings. Messages in input to Sign.
-- H_1,..., H_L (OPTIONAL), points of G1. The generators in input to Sign.
-- RevealedIndexes (OPTIONAL), vector of unsigned integers. Indexes of revealed messages.
+- disclosedIndexes (OPTIONAL), vector of unsigned integers in ascending
```

Comment:
if order matters should this be optional?

Reply:
Thank you very much for your feedback!! This can be confusing, but I think having it as REQUIRED will be like "forcing" the prover to disclose messages. So, I don't see a better alternative using the current syntax. I’ve added an `if not supplied it defaults to the empty array "()"` to try and make it clearer. 

In general, what I want to enforce is that if not supplied, no revealed messages can be supplied either. However, if it is supplied, it must be ordered.

Reply:
It seems to me like messages should come before disclosedIndexes in the argument list if only because of the optionality... if there are no signed messages then there are definitely no disclosedIndexes, but the same can't be said for the reverse.

Reply:
Yeah I agree with this ^

Reply:
Done!